website/client/components/shops/market/index.vue

```diff
@@ -23,223 +23,84 @@
-          span.dropdown-label {{ $t('sortBy') }}
-          b-dropdown(:text="$t(selectedSortItemsBy)", right=true)
-            b-dropdown-item(
-              v-for="sort in sortItemsBy",
-              @click="selectedSortItemsBy = sort",
-              :active="selectedSortItemsBy === sort",
-              :key="sort"
-            ) {{ $t(sort) }}
-
 
+      layout-section(:title="$t('items')")
+        div(slot="filters")
+          filter-dropdown(
+            :label="$t('sortBy')",
+            :initialItem="selectedSortItemsBy",
+            :items="sortItemsBy",
+            @selected="selectedSortItemsBy = $event"
+          )
+            span(slot="item", slot-scope="ctx")
+              span.text {{ $t(ctx.item.id) }}
       div(
         v-for="category in categories",
         v-if="viewOptions[category.identifier].selected && category.identifier !== 'equipment'"
       )
         h4 {{ category.text }}
-
-        div.items
-          shopItem(
-            v-for="item in sortedMarketItems(category, selectedSortItemsBy, searchTextThrottled, hidePinned)",
-            :key="item.key",
-            :item="item",
-            :emptyItem="false",
-            :popoverPosition="'top'",
-            @click="itemSelected(item)"
+          category-row(
+            :hidePinned="hidePinned",
+            :hideLocked="hideLocked",
+            :searchBy="searchTextThrottled",
+            :sortBy="selectedSortItemsBy.id",
+            :category="category"
           )
-            span(slot="popoverContent")
-              strong(v-if='item.key === "gem" && gemsLeft === 0') {{ $t('maxBuyGems') }}
-              h4.popover-content-title {{ item.text }}
-            template(slot="itemBadge", slot-scope="ctx")
-              countBadge(
-                v-if="item.showCount != false",
-                :show="userItems[item.purchaseType][item.key] != 0",
-                :count="userItems[item.purchaseType][item.key] || 0"
-              )
-              .badge.badge-pill.badge-purple.gems-left(v-if='item.key === "gem"')
-                | {{ gemsLeft }}
-              span.badge.badge-pill.badge-item.badge-svg(
-                :class="{'item-selected-badge': ctx.item.pinned, 'hide': !ctx.item.pinned}",
-                @click.prevent.stop="togglePinned(ctx.item)"
-              )
-                span.svg-icon.inline.icon-12.color(v-html="icons.pin")
 
           keys-to-kennel(v-if='category.identifier === "special"')
 
         div.fill-height
 
-      //- @TODO: Create new InventoryDrawer component and re-use in 'inventory/stable' component.
-      drawer(
-        :title="$t('quickInventory')"
-        :errorMessage="inventoryDrawerErrorMessage(selectedDrawerItemType)"
-      )
-        div(slot="drawer-header")
-          drawer-header-tabs(
-            :tabs="drawerTabs",
-            @changedPosition="tabSelected($event)"
+      inventoryDrawer(:showEggs="true", :showPotions="true")
+        template(slot="item", slot-scope="ctx")
```

Comment:
Same here for the slot. I don't think we customize the slot anywhere, so we should remove the slot and move the child elements to the component.

Reply:
Ah okay, I didn't see the second use of it :P That's perfectly fine then.